cmd/adapter/adapter.go

```diff
@@ -334,12 +356,35 @@ func main() {
 		klog.Fatalf("unable to install resource metrics API: %v", err)
 	}
 
+	config, err := cmd.Config()
+	if err != nil {
+		klog.Fatalf("unable to get configuration of custom metrics apiserver: %v", err)
+	}
+
+	apiServerConfig(config)
+	cmd.setupMetricsServer()
```

Comment:
To be able to shutdown both servers in case anything happens to one of them, it would be better to have them in run groups. You can take this implementation from kube-state-metrics as an example: https://github.com/kubernetes/kube-state-metrics/blob/master/main.go#L182-L203

Reply:
Isnt this line 
https://github.com/kubernetes-sigs/prometheus-adapter/pull/451/files#diff-4d0b81c92e53ab757dd6ce4dca5fdf332665e5e6d01e2e8465e517f5549886cdR385 effectively doing the same thing?  In this case I do not have start up of the other server as it is in the custom-metrics-apiserver

Reply:
I have implemented the run group for just one of them, the second one does accept a context. Let me know if this is useful, otherwise will revert.